### PR TITLE
Meal Calender added

### PR DESCRIPTION
**Pull request by ZSank**
# Pull Request

## Description

Please include a summary of the changes you have made. Highlight the key aspects of your project.
Added a calendar UI in the meal screen. 
User can select a date, and it will show at the top header text. 
Also, the arrow buttons can be used to navigate between the weeks. 

## Checklist

- [ ] My project includes a `README.md` file with a brief overview of the project, how to set it up, and any relevant information.
- [x] I have added a screenshot or screen recording showing the functionality of the app.
- [x] I have tested the project and confirmed it works as expected.

## Screenshots

If applicable, please add a screenshot or a screen recording to demonstrate the functionality of your app.

Thank you for your contribution!

### CODE REVIEW

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/meal_plan/CalenderDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+class CalendarDataSource {
+    val today: LocalDate
+        get() {
+            return LocalDate.now()
+        }
+
+
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/meal_plan/CalenderDataSource.kt]: Remove extra space.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+			fontWeight = FontWeight.ExtraBold
+		)
+		NavigationIconButton(
+			imageVector = Icons.Filled.ChevronLeft,
+			contentDescription = "Back",
+			onClick = onPrevClickListener
+		)
+		NavigationIconButton(
+			imageVector = Icons.Filled.ChevronRight,
+			contentDescription = "Next",
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt]: For now, either make it as null or get the string value from strings.xml

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+			onClick = onNextClickListener
+		)
+	}
+}
+
+@Composable
+fun NavigationIconButton(
+	imageVector: ImageVector,
+	contentDescription: String,
+	onClick: () -> Unit,
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt]: Move this function to utils/custom_view folder.

**Review by Siduchels — CHANGES_REQUESTED** [review]

**Review by aliarbaaz — APPROVED** [review]
Looks Good

**Review by Siduchels — APPROVED** [review]
Looks good.

**Pull request merged**